### ISSUE TITLE
enigma plugin: gpg redressing attack

### ISSUE THREAD

**Issue by hannob** [roundcube/roundcubemail#6450]
I want to report a security vulnerability in the way signed messages are displayed in the enigma plugin.

Signed messages are displayed with a green box in the message window. As this is space that is controlled by the sender of a mail it should be considered attacker controlled.

An attacker trying to fake a mail signature can simply rebuild that green box with HTML/CSS and add that to the top of the mail. (I skipped creating a PoC for this, as I already did this for several other mail clients and rebuilding things in HTML gets boring, but it should be straightforward obvious that this is possible.)

The information about a valid signature must not be displayed in attacker-controlled space. One way of fixing this is to put the signature info box above the mail headers. (This is what enigmail did after I reported it to them.)

**Comment by jensvoid** [comment]
Note that this issue also applies to the way S/MIME signatures are presented to the user by Roundcube/Enigma.

**Comment by alecpl** [comment]
Proposed solution: #6759.
![enigma-msg](https://user-images.githubusercontent.com/546788/57927553-a63a6c00-78ae-11e9-9c8c-1806339907c1.png)
![enigma-settings](https://user-images.githubusercontent.com/546788/57927556-a8042f80-78ae-11e9-9d65-1e53b8154480.png)

**Comment by lambdafu** [comment]
As an alternative, maybe consider opening attachments in their own tab or window. There is a huge benefit from having only one signature or encryption context per message.  Then you can also verify that the scope of that context is the whole email and mitigate wrapping attacks.

**Comment by MattyShires** [comment]
Has any action been done on this?

**Comment by alecpl** [comment]
No. There's a PR waiting for comments and some more considerations if it's a appropriate way of fixing the issue.

**Comment by JohnRDOrazio** [comment]
Wouldn't it be possible to place the message box in the same space as the "This message contains attached PGP key(s)" message box? i.e. https://github.com/roundcube/roundcubemail/blob/dec1d668ed37f8d4662eb37a37488f3fcce5fa43/plugins/enigma/lib/enigma_ui.php#L1098

**Comment by JohnRDOrazio** [comment]
I tested my proposed solution on my own Roundcube instance, and it worked, so I've made PR #10007 to address this. The results on my live Roundcube instance:

<img width="622" height="252" alt="Image" src="https://github.com/user-attachments/assets/f8fec943-e935-491b-9dc9-5721dd3cc9ed" />
